homeassistant/components/homewizard/sensor.py

```diff
@@ -383,11 +376,149 @@ async def async_setup_entry(
+    if coordinator.data.data.external_devices is not None:
+        for (unique_id, device) in coordinator.data.data.external_devices.items():
+
+            description: SensorEntityDescription | None = None
+
+            if device.meter_type == ExternalDevice.DeviceType.GAS_METER:
+                description = SensorEntityDescription(
+                    key="Gas meter",
+                    name="Total gas",
+                    device_class=SensorDeviceClass.GAS,
+                    state_class=SensorStateClass.TOTAL_INCREASING,
+                )
+
+            if device.meter_type == ExternalDevice.DeviceType.HEAT_METER:
+                description = SensorEntityDescription(
+                    key="Gas meter",
+                    name="Total energy consumed",
+                    device_class=SensorDeviceClass.ENERGY,
+                    state_class=SensorStateClass.TOTAL_INCREASING,
+                )
+
+            if device.meter_type == ExternalDevice.DeviceType.WARM_WATER_METER:
+                description = SensorEntityDescription(
+                    key="Warm water meter",
+                    name="Total water",
+                    device_class=SensorDeviceClass.WATER,
+                    state_class=SensorStateClass.TOTAL_INCREASING,
+                )
+
+            if device.meter_type == ExternalDevice.DeviceType.WATER_METER:
+                description = SensorEntityDescription(
+                    key="Water meter",
+                    name="Total water",
+                    device_class=SensorDeviceClass.WATER,
+                    state_class=SensorStateClass.TOTAL_INCREASING,
+                )
+
+            if device.meter_type == ExternalDevice.DeviceType.INLET_HEAT_METER:
+                description = SensorEntityDescription(
+                    key="Inlet Heat meter",
+                    name="Total energy",
+                    device_class=SensorDeviceClass.ENERGY,
+                    state_class=SensorStateClass.TOTAL_INCREASING,
+                )
+
+            if description:
+                entities.append(
+                    HomeWizardExternalSensorEntity(coordinator, description, unique_id)
+                )
+
+    async_add_entities(entities)
+
+
+class HomeWizardExternalSensorEntity(HomeWizardEntity, SensorEntity):
+    """Representation of externally connected HomeWizard Sensor."""
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        description: SensorEntityDescription,
+        device_unique_id: str,
+    ) -> None:
+        """Initialize Externally connected HomeWizard Sensors."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{DOMAIN}_{device_unique_id}"
+        self._device_id = device_unique_id
+        self.entity_description = description
+        self._attr_native_unit_of_measurement = self._get_native_unit_of_measurement()
+
+        if self._attr_device_info is not None:
```

Comment:
This _is_ set in `super().__init__..` but MyPy does not not checking this. This feels a bit ugly.